content/academy/apify_platform/deploying/inputs_outputs.md

```diff
@@ -0,0 +1,193 @@
+
+```JSON
+{
+    "numbers": [5, 5, 5, 5]
+}
+```
+
+Then we can modify our code to grab the input and use it to generate a solution which is logged to the console.
+
+```JavaScript
+// index.js
+import Apify from 'apify';
+
+const { numbers } = await Apify.getInput();
+
+const addAllNumbers = (...nums) => nums.reduce((total, curr) => (total += curr));
+
+const solution = addAllNumbers(...numbers);
+
+console.log(solution);
+```
+
+Cool! When we run `node index.js`, we see **20**.
+
+### Accepting input without the Apify SDK
+
+Alternatively, when writing in a language other than JavaScript, we can create our own `get_input()` function which utilizes the Apify API. For this example, we are using the [Apify Client]({{@link apify_platform/getting_started/apify_client.md}}) for Python to access the API.
+
+```Python
+# index.py
+from apify_client import ApifyClient
+from os import environ
+import json
+
+client = ApifyClient(token='YOUR_TOKEN')
+
+# If being run on the platform, the "APIFY_IS_AT_HOME" environment variable
+# will be "1". Otherwise, it will be undefined/None
+def is_on_apify ():
+    if (environ.get('APIFY_IS_AT_HOME')):
+        return True
+    return False
+
+# Get the input
+def get_input ():
+    if (not is_on_apify()):
+        input = open('./apify_storage/key_value_stores/default/INPUT.json')
+        return json.load(input)
+
+    kv_store = client.key_value_store(environ.get('APIFY_DEFAULT_KEY_VALUE_STORE_ID'))
+    kv_store.get_record('INPUT', format='json')
+
+def add_all_numbers (nums):
+    total = 0
+
+    for num in nums:
+        total += num
+
+    return total
+
+solution = add_all_numbers(get_input()['numbers'])
+
+print(solution)
+```
+
+> For a better understanding of API endpoints reading and modifying key-value stores, check the [official API reference](https://docs.apify.com/api/v2#/reference/key-value-stores).
+
+## [](#writing-output) Writing output
+
+Similarly to reading input, you can write the actor's output either by using the Apify SDK in Node.js, or by manually writing a utility function to do so.
+
+### Writing output with the Apify SDK
+
+In the SDK, we can write to the key-value store with the `Apify.setValue()` function. Let's go ahead and write the solution of the `addAllNumbers()` function to the key-value store using it:
+
+```JavaScript
+// index.js
+import Apify from 'apify';
+
+const { numbers } = await Apify.getInput();
+
+const addAllNumbers = (...nums) => nums.reduce((total, curr) => (total += curr));
+
+const solution = addAllNumbers(...numbers);
+
+await Apify.setValue('OUTPUT', { solution });
```

Comment:
I know that a KVS seems like the obvious option here, but could we do a Dataset instead? I'm asking because output schema (pretty output in app) is only working with datasets now.

Reply:
I agree it should be dataset, just used the KVStore because Lukas used KVStore in his Rust example which I translated to JS, then translated that to Python haha